File: HovyFridge/app/src/main/java/com/niklyadov/hovyfridge/data/entities/Fridge.kt
package com.niklyadov.hovyfridge.data.entities

import androidx.room.*
import com.google.gson.annotations.SerializedName
import com.niklyadov.hovyfridge.data.entities.relations.FridgeWithProducts

@Entity
data class Fridge (
    @PrimaryKey
    @SerializedName("id")
    val id : Int,

    @SerializedName("is_deleted")
    val isDeleted : Boolean = false,

    @ColumnInfo(name = "name")
    @SerializedName("name")
    var name : String,
)
{
    @Ignore
    @SerializedName("products")
    var products: List<Product> = listOf()
}

File: HovyFridge.Web/HovyFridge/Data/Entity/Fridge.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace HovyFridge.Data.Entity
{
    public class Fridge : Entity
    {
        public string Name { get; set; } = string.Empty;

        public string? Description { get; set; }

        public List<Product> Products { get; set; } = new List<Product>();

        [NotMapped]
        public int ProductsAmount { get; set; }
        public long? OwnerId { get; set; }
    }
}
